dom.go

```diff
@@ -717,21 +720,63 @@ func Rerender(c Component) {
 	if c == nil {
 		panic("vecty: Rerender illegally called with a nil Component argument")
 	}
-	prevRender := c.Context().prevRender
-	if prevRender == nil {
-		panic("vecty: Rerender invoked on Component that has never been rendered")
-	}
 	if c.Context().unmounted {
 		return
 	}
-	nextRender, skip, pendingMounts := renderComponent(c, prevRender)
-	if skip {
-		return
+	batch.add(c)
+}
+
+// batchRenderer handles component re-renders by queueing and deduplicating
+// them, to be rendered on the next animation frame (via requestAnimationFrame).
+type batchRenderer struct {
+	// batch contains the list of pending components to render.
+	batch []Component
+	// scheduled tracks whether a batch has been scheduled for processing.
+	scheduled bool
+}
+
+// add a Component to the pending batch.
+func (b *batchRenderer) add(c Component) {
+	b.batch = append(b.batch, c)
+	// If we're not already scheduled for a render batch, request a render on
+	// the next frame.
+	if !b.scheduled {
+		b.scheduled = true
+		RequestAnimationFrame(b.render)
+	}
+}
+
+// render the pending batch.
+func (b *batchRenderer) render(float64) {
+	// Drain the current batch.
+	var pending []Component
+	pending, b.batch = append(pending, b.batch...), nil
```

Comment:
A better way to write this would be:

```Go
pending := b.batch
b.batch = nil
```

Because that avoids copying the entire slice.